messenger/message-recorder.rst

```diff
@@ -0,0 +1,120 @@
+.. index::
+    single: Messenger; Record messages; Transaction messages
+
+Transactional Messages: Handle Events After CommandHandler Is Done
+==================================================================
+
+A message handler can ``dispatch`` new messages during execution, to either the same or
+a different bus (if the application has `multiple buses </messenger/multiple_buses>`_).
+Any errors or exceptions that occur during this process can have unintended consequences,
+such as:
+
+- If using the ``DoctrineTransactionMiddleware`` and a dispatched message to the same bus
+  and an exception is thrown, then any database transactions in the original handler will
+  be rolled back.
+- If the message is dispatched to a different bus, then the dispatched message can still
+  be handled even if the original handler encounters an exception.
+
+An Example ``SignUpUser`` Process
+---------------------------------
+
+Let's take the example of an application with both a *command* and an *event* bus. The application
+dispatches a command named ``SignUpUser`` to the command bus. The command is handled by the
+``SignUpUserHandler`` which creates a ``User`` object, stores that object to a database and
+dispatches a ``UserSignedUp`` event to the event bus.
+
+There are many subscribers to the ``UserSignedUp`` event, one subscriber may send
+a welcome email to the new user. We are using the ``DoctrineTransactionMiddleware``
+to wrap all database queries in one database transaction.
+
+**Problem 1:** If an exception is thrown when sending the welcome email, then the user
+will not be created because the ``DoctrineTransactionMiddleware`` will rollback the
+Doctrine transaction, in which the user has been created.
```

Comment:
I thought (from your description earlier) that because `UserSignedUp` is sent to a different bus, if it throws an exception, it will not affect the transaction wrapping the command bus... and so that transaction (the one that saves the user) will still be committed. I'm sure i'm wrong - but I'm getting a bit lost :)

Reply:
Nop, that's the point of this `DispatchAfterCurrentBusStamp` :)